datacube/storage/storage.py

```diff
@@ -159,7 +159,8 @@ def read_from_source(source, dest, dst_transform, dst_nodata, dst_projection, re
                 _LOG.debug('Failed Read: %s', src, exc_info=1)
                 return
             dest = dest[offset[0]:offset[0] + tmp.shape[0], offset[1]:offset[1] + tmp.shape[1]]
-            numpy.copyto(dest, tmp, where=(tmp != src.nodata))
+            where_not_nodata = (tmp != src.nodata) if not numpy.isnan(src.nodata) else ~numpy.isnan(tmp)
```

Comment:
double negative in a variable name is a bit confusing, how about `valid_data`/`valid_mask`?